### ISSUE TITLE
Label missing in validation meta file

### ISSUE THREAD

**Issue by SaoYear** [DCASE-REPO/DESED_task#32]
I found some audio clips in the validation set that do not have labels.

Can I safely delete them from the tsv files?

List of these files:
```
                            filename  onset  offset event_label
4051    YU0Cg_t_3TdI_30.000_40.000.wav    NaN     NaN         NaN
4059     Y70CtvdsGQf4_2.000_12.000.wav    NaN     NaN         NaN
4074    Y-jLujkP1aaw_30.000_40.000.wav    NaN     NaN         NaN
4093  YY09tBEoLETY_440.000_450.000.wav    NaN     NaN         NaN
4114    Y-YwPSwhe2jo_30.000_40.000.wav    NaN     NaN         NaN
4115    Y-RLILmQmubM_30.000_40.000.wav    NaN     NaN         NaN
4156    YjF73O9J4KEI_30.000_40.000.wav    NaN     NaN         NaN
4161  YuXDi40QfH9E_320.000_330.000.wav    NaN     NaN         NaN
4171     Y0IxyUtHw6Ec_0.000_10.000.wav    NaN     NaN         NaN
4187    Y-4pmCrSdMhg_30.000_40.000.wav    NaN     NaN         NaN
4188  YQb6N3HvbmgQ_580.000_590.000.wav    NaN     NaN         NaN
4222  Y09kqMqn5rWs_260.000_270.000.wav    NaN     NaN         NaN
4223    YzDULhAxsJfY_60.000_70.000.wav    NaN     NaN         NaN
4228    YR3_Gqgl-NYE_50.000_60.000.wav    NaN     NaN         NaN
4238    Y86owBlJa8f0_24.000_34.000.wav    NaN     NaN         NaN
```

I realized that these samples could testify the FPs for the models. But it seems that they would lead to some bugs in my code after I update my pandas package (not sure). Did anyone have the same issue?

UPDATE: 
My solution:
in `desed_task\dataio\datasets.py`

line 141:
put `labels_df = process_labels(labels_df, onset_s, offset_s)` into the following if-else:
```# check if labels exists:
        if not len(labels_df):
            max_len_targets = self.encoder.n_frames
            strong = torch.zeros(max_len_targets, len(self.encoder.labels)).float()
        else:
            labels_df = process_labels(labels_df, onset_s, offset_s)
            strong = self.encoder.encode_strong_df(labels_df)
            strong = torch.from_numpy(strong).float()
```
Dont konw yet if it is the right thing to do, cuz my test for the provided baseline.ckpt is higher than the given value (student: 33.431%, 53.177%; teacher: 35.020%, 54.955%).

Modifications:
1. Use torchmetric.F1Score to replace the F1 function in pytorch-lightning;
2. Replace all df.append by pd.concat([df1, df2], ignore_index=True/False)

**Comment by RonFrancesca** [comment]
HI @SaoYear, 

sorry for the late answer. 
As you commented, some of the validation files do not have labels for onset, offset and event_label.

They need to be dropped and this is handled in this [pull request](https://github.com/DCASE-REPO/DESED_task/pull/37/commits), already merged to the master branch. 

So the code is not correctly considering the validation set and you should be able to get the same results as reported in the [README file](https://github.com/DCASE-REPO/DESED_task/tree/master/recipes/dcase2022_task4_baseline). 

In order to have the right results, you should use psds_eval >= 0.4.0. 

Hope this helps.
